src/serialize.cpp

```diff
@@ -181,21 +183,29 @@ std::vector<uint8_t> serialize_primitive_array(const sparrow::primitive_array<T>
         batch_builder.Finish(batch_message_offset);
 
         // III - Append the RecordBatch message to the final buffer
-        uint32_t batch_meta_len = batch_builder.GetSize(); // Get the size of the batch metadata
-        int64_t aligned_batch_meta_len = align_to_8(batch_meta_len); // Calculate the padded length
+        const uint32_t batch_meta_len = batch_builder.GetSize(); // Get the size of the batch metadata
+        const int64_t aligned_batch_meta_len = align_to_8(batch_meta_len); // Calculate the padded length
 
-        size_t current_size = final_buffer.size(); // Get the current size (which is the end of the Schema message)
+        const size_t current_size = final_buffer.size(); // Get the current size (which is the end of the Schema message)
         // Resize the buffer to append the new message
         final_buffer.resize(current_size + sizeof(uint32_t) + aligned_batch_meta_len + body_len);
         uint8_t* dst = final_buffer.data() + current_size; // Get a pointer to where the new message will start
 
         // Write the 4-byte metadata length for the RecordBatch message
-        *(reinterpret_cast<uint32_t*>(dst)) = batch_meta_len;
+        memcpy(dst, &batch_meta_len, sizeof(batch_meta_len));
         dst += sizeof(uint32_t);
         // Copy the RecordBatch metadata into the buffer
         memcpy(dst, batch_builder.GetBufferPointer(), batch_meta_len);
         // Add padding to align the body to an 8-byte boundary
-        memset(dst + batch_meta_len, 0, aligned_batch_meta_len - batch_meta_len);
+        if (static_cast<size_t>(aligned_batch_meta_len) >= batch_meta_len)
```

Comment:
```
src/serialize.cpp:200:13 [modernize-use-integer-sign-comparison]
comparison between 'signed' and 'unsigned' integers
```

Still complaining even after the cast...
The thing is that we're kind of constrained on some of the functions from the generated headers in arrow/flatbuffers, thus mixing up unsigned and signed integers in general in this codebase...